custom/icds_reports/static/js/angular-services/locations.service.js

```diff
@@ -305,6 +305,11 @@ window.angular.module('icdsApp').factory('locationsService', ['$http', '$locatio
         },
         
         resetLevelsBelow : function(level, vm) {
+            if (vm.selectedLocationLevel > level &&
+                vm.selectedLocations[level] !== undefined &&
+                vm.selectedLocations[level] !== null) {
```

Comment:
do you need to explicitly check against `undefined` and `null` and not just `vm.selectedLocations[level]`?

Reply:
hey, No explicit check for null and undefined.